src/prepare.js

```diff
@@ -94,6 +93,12 @@ function prepareElement(element, errorHandler, context) {
     return Promise.resolve([consumerFunc(value), context]);
   }
 
+  if (
+    typeof type === 'object' &&
+    type.$$typeof.toString() === 'Symbol(react.forward_ref)'
+  ) {
+    return Promise.resolve([type.render(props), context]);
```

Comment:
```suggestion
    return Promise.resolve([type.render(props, element.ref), context]);
```

So we ensure we pass in the ref.

Reply:
Looks good. Is there any documentation on the type.render method, f. ex. to see which parameters it accepts?

Reply:
Ish

All this stuff is react internals, and they doesn't have "official" apis like the normal react package.

You can understand a bit more by looking at the source tho:
https://github.com/facebook/react/blob/master/packages/react/src/forwardRef.js#L13